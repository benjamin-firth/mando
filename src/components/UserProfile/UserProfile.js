import React from 'react';
import { NavLink } from 'react-router-dom';

const UserProfile = ({ currentUser, changePage, handleImgClick }) => {
  return (
    <section className='profile-box'>
      <h3>Name: {currentUser.name}</h3>
      <p>"{currentUser.quote}"</p>
      <h4>Experience Level: {currentUser.rank}</h4>
      <NavLink id='welcome-form' onClick={() => changePage('WelcomeForm', {}, 'hide')} exact to={'/'}>Log Out</NavLink>
      <NavLink id='nav-link2' exact to={'/movie-page'} onClick={() => handleImgClick()}>Home</NavLink>
    </section>
  )
}

export default UserProfile;
