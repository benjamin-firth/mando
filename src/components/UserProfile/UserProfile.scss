@import '../App/App.scss';

.profile-box {
  align-self: flex-start;
  background-color: #1C628E;
  display: flex;
  flex-direction: column;
  font-size: 20px;
  left: 0;
  padding: 0 0 15px 0;
  position: absolute;
  right: 0;
  z-index: 1;
}

#welcome-form,
#nav-link2 {
  background-color: black;
  border-radius: 5px;
  color: white;
  margin: 10px;
  padding: 5px 10px;
  text-decoration: none;
}
