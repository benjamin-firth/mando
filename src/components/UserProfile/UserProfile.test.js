import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile.js'

describe('UserProfile', () => {
  let wrapper;
  let mockProps;
  let mockChangePage;
  let mockHandleImgClick;

  beforeEach(() => {
    mockProps = {
      name: 'John Adams',
      quote: 'You send them to me!',
      rank: 'Master'
    }
    mockChangePage = jest.fn();
    mockHandleImgClick = jest.fn();
    wrapper = shallow(<UserProfile currentUser={mockProps} changePage={mockChangePage} handleImgClick={mockHandleImgClick}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire changePage when navLink is clicked', () => {
    wrapper.find('#welcome-form').simulate('click');

    expect(mockChangePage).toHaveBeenCalledWith('WelcomeForm', {}, 'hide');
  });

  it('should fire handleImgClick when other navLink is clicked', () => {
    wrapper.find('#nav-link2').simulate('click');

    expect(mockHandleImgClick).toHaveBeenCalled();
  })
})
