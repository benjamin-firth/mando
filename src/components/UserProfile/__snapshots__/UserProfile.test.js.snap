// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<section
  className="profile-box"
>
  <h3>
    Name: 
    John Adams
  </h3>
  <p>
    "
    You send them to me!
    "
  </p>
  <h4>
    Experience Level: 
    Master
  </h4>
  <NavLink
    exact={true}
    id="welcome-form"
    onClick={[Function]}
    to="/"
  >
    Log Out
  </NavLink>
  <NavLink
    exact={true}
    onClick={[Function]}
    to="/movie-page"
  >
    Home
  </NavLink>
</section>
`;
